vendor/github.com/ethereum/go-ethereum/common/message/message.go

```diff
@@ -0,0 +1,65 @@
+package message
+
+// Direction defines a int type to indicate a message as either incoming or outgoing.
+type Direction int
+
+// consts of all message direction values.
+const (
+	IncomingMessage Direction = iota << 2
+	OutgoingMessage
+)
+
+// String returns the representation of giving direction.
+func (d Direction) String() string {
+	switch d {
+	case IncomingMessage:
+		return "IncomingMessage"
+	case OutgoingMessage:
+		return "OutgoingMessage"
+	}
+
+	return "MessageDirectionUnknown"
+}
+
+// Status defines a int type to indicate different status value of a
+// message state.
+type Status int
+
+// consts of all message delivery status.
+const (
+	PendingStatus Status = iota << 3
```

Comment:
Question regarding `iota` usage for both `Status` and `Direction` types: why using bit shift of `iota` here?

It compiles to:

```go
        PendingStatus Status = iota << 3  // 0 << 3 = 0
	QueuedStatus   // 1 << 3 = 8
 	CachedStatus   // 2 << 3 = 16
 	SentStatus     // 3 << 3 = 24
 	ExpiredStatus  // 4 << 3 = 32
        ....
```

Using bitshift with iota usually implies using power of two and typically used as `1 << iota` (not `iota << 1` or `iota << 2`), and useful for encompassing values int combined ones, like `var CombinedStatus = SentStatus | ResentStatus`. But it seems both `Status` and `Direction` are never used that way, and are simple autoincrementing constants.

Even more, there is a peril with having first constant in a group of value 0. Zero is a default "zero value" for the int type, so any unitialized variable of type `Direction` will automatically have `IncomingMessage` value, which is not technically true and may lead to subtle bugs. This is avoided usually with either having `InitialStatus` const or by using `iota + 1` (or `1 << iota + 1` in case of bitshifted iota).

So unless I'm missing something important here, I think it would be better to change those two const groups to use simple `iota + 1` - less confusing, no zero value issue and easier to understand.

Reply:
Hmm, agreed on this, will make the change.